pkg/loop/internal/relayer/pluginprovider/contractreader/contract_reader_test.go

```diff
@@ -434,36 +435,135 @@ func (f *fakeContractReader) SetBatchLatestValues(batchCallEntry BatchCallEntry)
 }
 
 func (f *fakeContractReader) GetLatestValue(_ context.Context, readIdentifier string, confidenceLevel primitives.ConfidenceLevel, params, returnVal any) error {
+	_, isValue := returnVal.(*values.Value)
+
+	if !isValue {
```

Comment:
can't you just wrap the result before returning it instead of encoding it?  I feel like there's a lot of duplication otherwise.

Reply:
true, refactored to deduplicate